int32 autoaim_mode